src/main/java/com/pingcap/tikv/operation/SchemaInfer.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2017 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pingcap.tikv.operation;
+
+
+import com.pingcap.tikv.expression.aggregate.Sum;
+import com.pingcap.tikv.meta.TiSelectRequest;
+import com.pingcap.tikv.types.FieldType;
+import com.pingcap.tikv.types.StringBaseType;
+import com.pingcap.tikv.types.VarCharType;
+import lombok.Data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * SchemaInfer extract row's type after query is executed.
+ * It is pretty rough version for. Optimization is on the way.
+ * The problem we have right now is that TiDB promote Sum to Decimal which is
+ * not compatible with column's type.
+ * The solution we come up with right now is use record column's type ad finalFieldType
+ * and build another list recording TiExpr's type as fieldType for row reading.
+ * Once we finish row reading, we first check each element in fieldType and finalFieldType share
+ * the same type or not. If yes, no need for casting. If no, casting is needed here.
+ */
+public class SchemaInfer {
+
+    public static TiFieldType create(TiSelectRequest tiSelectRequest) {
+        return new TiFieldType(tiSelectRequest);
+    }
+
+    @Data
+    public static class TiFieldType {
```

Comment:
Naming is not intuitive. Shouldn't be TiCoprocessorRequest or something similar?